stable/kubernetes-dashboard/templates/clusterrole-readonly.yaml

```diff
@@ -0,0 +1,160 @@
+{{- if and .Values.rbac.create (not .Values.rbac.clusterAdminRole) }}
+apiVersion: rbac.authorization.k8s.io/v1beta1
+kind: ClusterRole
+metadata:
+  labels:
+    app: {{ template "kubernetes-dashboard.name" . }}
+    chart: {{ template "kubernetes-dashboard.chart" . }}
+    heritage: {{ .Release.Service }}
+    release: {{ .Release.Name }}
+  name: "{{ template "kubernetes-dashboard.fullname" . }}-readonly"
+  namespace: {{ .Release.Namespace }}
+rules:
+  # Allow Dashboard to get, update and delete Dashboard exclusive secrets.
+  - apiGroups:
+      - ""
+    resources:
+      - secrets
+    resourceNames:
+      - kubernetes-dashboard-key-holder
+      - {{ template "kubernetes-dashboard.fullname" . }}
+    verbs:
+      - get
+      - update
+      - delete
+
+  # Allow Dashboard to get and update 'kubernetes-dashboard-settings' config map.
+  - apiGroups:
+      - ""
+    resources:
+      - configmaps
+    resourceNames:
+      - kubernetes-dashboard-settings
+    verbs:
+      - get
+      - update
+
+  - apiGroups:
+      - ""
+    resources:
+      - configmaps
+      - endpoints
+      - persistentvolumeclaims
+      - pods
+      - replicationcontrollers
+      - replicationcontrollers/scale
+      - serviceaccounts
+      - services
+      - nodes
+      - persistentvolumeclaims
+      - persistentvolumes
```

Comment:
I'm not sure persistent volumes never contain secrets... They are intended to be cluster admin only usually so I'm sure if its proper or not for them to have secrets embedded in them.

Its possible any of these other objects also could improperly contain secrets too though... This maybe should be called out explicitly in the docs that this feature should be used with extreme care.